src/vm/eventpipe.cpp

```diff
@@ -185,6 +184,10 @@ bool EventPipeProviderCallbackDataQueue::TryDequeue(EventPipeProviderCallbackDat
 void EventPipe::Initialize()
 {
     STANDARD_VM_CONTRACT;
+    PRECONDITION(!s_tracingInitialized);
+
+    if (s_tracingInitialized)
```

Comment:
Is it safe to check this without taking the lock?

Reply:
Need to review all places where we use the flag.

Reply:
Does your comment mean this function can be entered more than once?  If this function can be entered multiple times, won't `s_tracingInitialized` be true on later entries, thus violating the precondition?

Reply:
It should not be entered more than once, but I wanted to test it, as Shutdown was called twice.

Reply:
Nvm. `s_tracingInitialized` is volatile, so I'm guessing this is fine.